{"ygPerms":{"resourceCapabilityList":[{"resourceType":"GROUP","capabilities":[{"name":"READ"},{"name":"JOIN"}]},{"resourceType":"PHOTO","capabilities":[]},{"resourceType":"FILE","capabilities":[]},{"resourceType":"MEMBER","capabilities":[]},{"resourceType":"LINK","capabilities":[]},{"resourceType":"CALENDAR","capabilities":[]},{"resourceType":"DATABASE","capabilities":[]},{"resourceType":"POLL","capabilities":[]},{"resourceType":"MESSAGE","capabilities":[{"name":"READ"}]},{"resourceType":"PENDING_MESSAGE","capabilities":[]},{"resourceType":"ATTACHMENTS","capabilities":[{"name":"READ"}]},{"resourceType":"PHOTOMATIC_ALBUMS","capabilities":[]},{"resourceType":"MEMBERSHIP_TYPE","capabilities":[]},{"resourceType":"POST","capabilities":[{"name":"READ"}]},{"resourceType":"PIN","capabilities":[]}],"groupUrl":"groups.yahoo.com","intlCode":"us"},"comscore":"pageview_candidate","ygData":{"userId":137285340,"authorName":"Gordon Mohr","from":"Gordon Mohr &lt;gojomo@...&gt;","profile":"gojomo","replyTo":"LIST","senderId":"nlRylPFm-609k3-oCX01bLjsw78dUyU2D5WH-XYNWWxjPxjt2L4rGjw9FyyH_o7TxY4uYJb00hVhysPuJ28thjd688MK5Eo","spamInfo":{"isSpam":false,"reason":"0"},"subject":"Re: [archive-crawler] java.util.NoSuchElementException","postDate":"1175808124","msgId":4057,"canDelete":false,"contentTrasformed":false,"systemMessage":false,"headers":{"messageIdInHeader":"PDQ2MTU2ODdDLjYwMTAzMDBAYXJjaGl2ZS5vcmc+","inReplyToHeader":"PGV2M2ZpbStwNDF2QGVHcm91cHMuY29tPg==","referencesHeader":"PGV2M2ZpbStwNDF2QGVHcm91cHMuY29tPg=="},"prevInTopic":4054,"nextInTopic":0,"prevInTime":4056,"nextInTime":4058,"topicId":4030,"numMessagesInTopic":4,"msgSnippet":"... There appear to be 3 separate exceptions below, and none are problems I recognize. ... This is actually an exception in a background thread of the ","rawEmail":"Return-Path: &lt;gojomo@...&gt;\r\nX-Sender: gojomo@...\r\nX-Apparently-To: archive-crawler@yahoogroups.com\r\nReceived: (qmail 26395 invoked from network); 5 Apr 2007 21:20:25 -0000\r\nReceived: from unknown (66.218.67.35)\n  by m45.grp.scd.yahoo.com with QMQP; 5 Apr 2007 21:20:25 -0000\r\nReceived: from unknown (HELO mail.archive.org) (207.241.233.246)\n  by mta9.grp.scd.yahoo.com with SMTP; 5 Apr 2007 21:20:25 -0000\r\nReceived: from localhost (localhost [127.0.0.1])\n\tby mail.archive.org (Postfix) with ESMTP id A8EF7141C6E2B\n\tfor &lt;archive-crawler@yahoogroups.com&gt;; Thu,  5 Apr 2007 14:19:45 -0700 (PDT)\r\nReceived: from mail.archive.org ([127.0.0.1])\n\tby localhost (mail.archive.org [127.0.0.1]) (amavisd-new, port 10024)\n\twith LMTP id 12534-02-29 for &lt;archive-crawler@yahoogroups.com&gt;;\n\tThu, 5 Apr 2007 14:19:45 -0700 (PDT)\r\nReceived: from [192.168.1.203] (c-76-102-230-209.hsd1.ca.comcast.net [76.102.230.209])\n\tby mail.archive.org (Postfix) with ESMTP id 3FBA5141BC517\n\tfor &lt;archive-crawler@yahoogroups.com&gt;; Thu,  5 Apr 2007 14:19:45 -0700 (PDT)\r\nMessage-ID: &lt;4615687C.6010300@...&gt;\r\nDate: Thu, 05 Apr 2007 14:22:04 -0700\r\nUser-Agent: Thunderbird 1.5.0.10 (X11/20070306)\r\nMIME-Version: 1.0\r\nTo: archive-crawler@yahoogroups.com\r\nReferences: &lt;ev3fim+p41v@...&gt;\r\nIn-Reply-To: &lt;ev3fim+p41v@...&gt;\r\nContent-Type: text/plain; charset=ISO-8859-1; format=flowed\r\nContent-Transfer-Encoding: 7bit\r\nX-Virus-Scanned: Debian amavisd-new at archive.org\r\nX-eGroups-Msg-Info: 1:0:0:0\r\nFrom: Gordon Mohr &lt;gojomo@...&gt;\r\nSubject: Re: [archive-crawler] java.util.NoSuchElementException\r\nX-Yahoo-Group-Post: member; u=137285340; y=v90tq97ulx3Nb9KiG5mgXuMm1wBYfA0NhMVD4RdD75R5\r\nX-Yahoo-Profile: gojomo\r\n\r\nanitabidari wrote:\n&gt; Iam using heretrix 1.10.2, and am seeing the below exception. Any idea\n&gt; why this is occuring? Appears to be a Heretrix bug. Is there a version\n&gt; of Heretrix available that has a fix for this?\n\nThere appear to be 3 separate exceptions below, and none are problems I \nrecognize.\n\n&gt; \n&gt; &lt;Cleaner name=&quot;Cleaner-1&quot;/&gt; caught exception:\n&gt; java.util.NoSuchElementException\n&gt; java.util.NoSuchElementException\n&gt; at java.util.TreeMap.key(TreeMap.java:433)\n&gt; at java.util.TreeMap.firstKey(TreeMap.java:287)\n&gt; at\n&gt; com.sleepycat.je.cleaner.FileProcessor$LookAheadCache.nextOffset(FileProcessor.j&#92;\n&gt; ava:1147)\n&gt; at\n&gt; com.sleepycat.je.cleaner.FileProcessor.processLN(FileProcessor.java:525)\n&gt; at\n&gt; com.sleepycat.je.cleaner.FileProcessor.processFile(FileProcessor.java:481)\n&gt; at\n&gt; com.sleepycat.je.cleaner.FileProcessor.doClean(FileProcessor.java:235)\n&gt; at\n&gt; com.sleepycat.je.cleaner.FileProcessor.onWakeup(FileProcessor.java:143)\n&gt; at\n&gt; com.sleepycat.je.utilint.DaemonThread.run(DaemonThread.java:189)\n&gt; at java.lang.Thread.run(Thread.java:595)\n&gt; Continuing\n\nThis is actually an exception in a background thread of the \nBerkeleyDB-JE library we use. I haven&#39;t seen this before. Have you seen \nit more than once?\n\n\n&gt; Exception in thread &quot;waker for\n&gt; org.archive.crawler.admin.CrawlJob$MBeanCrawlController@19f03d7&quot;\n&gt; java.util.NoSuchElementException\n&gt; at java.util.TreeMap.key(TreeMap.java:433)\n&gt; at java.util.TreeMap.firstKey(TreeMap.java:287)\n&gt; at java.util.TreeSet.first(TreeSet.java:407)\n&gt; at\n&gt; java.util.Collections$SynchronizedSortedSet.first(Collections.java:1749)\n&gt; at\n&gt; org.archive.crawler.frontier.WorkQueueFrontier.wakeQueues(WorkQueueFrontier.java&#92;\n&gt; :788)\n&gt; at\n&gt; org.archive.crawler.frontier.WorkQueueFrontier$WakeTask.run(WorkQueueFrontier.ja&#92;\n&gt; va:88)\n&gt; at java.util.TimerThread.mainLoop(Timer.java:512)\n&gt; at java.util.TimerThread.run(Timer.java:462)\n\nThis is an exception in a Heritrix frontier background thread for waking \nqueues whose &#39;politeness slumber&#39; has finished. It is also unfamiliar. \nIt is curious that the exception comes from the same TreeMap method, \nthough this structure is unrelated to the BDB-JE &#39;Cleaner&#39; structures \nabove.\n\n&gt; Exception in thread &quot;ToeThread #79:\n&gt; http://tickets.aol.com/venue.adp?referrer=http%3a%2f%2ftickets%2eaol%2ecom%2f&vi&#92;\n&gt; d=106714&quot;\n&gt; java.lang.IncompatibleClassChangeError\n&gt; at\n&gt; java.util.ListResourceBundle.handleGetObject(ListResourceBundle.java:109)\n&gt; at java.util.ResourceBundle.getObject(ResourceBundle.java:319)\n&gt; at java.util.ResourceBundle.getString(ResourceBundle.java:285)\n&gt; at java.util.logging.Level.getLocalizedName(Level.java:224)\n&gt; at\n&gt; org.archive.util.OneLineSimpleLogger.format(OneLineSimpleLogger.java:85)\n&gt; at java.util.logging.StreamHandler.publish(StreamHandler.java:179)\n&gt; at\n&gt; java.util.logging.ConsoleHandler.publish(ConsoleHandler.java:88)\n&gt; at java.util.logging.Logger.log(Logger.java:428)\n&gt; at java.util.logging.Logger.doLog(Logger.java:450)\n&gt; at java.util.logging.Logger.log(Logger.java:539)\n&gt; at org.archive.crawler.framework.ToeThread.run(ToeThread.java:172)\n\nThis is really odd: an IncompatibleClassChangeError is pretty exotic and \ncertainly shouldn&#39;t be triggered by as simple of a logging operation as\nat org.archive.crawler.framework.ToeThread.run(ToeThread.java:172). \n(Beyond editting the system JARs on disk while Java was running, I&#39;m not \nsure I&#39;d know how to trigger this Exception on purpose.)\n\nTaken together, these three unique and hard-to-explain exceptions make \nme suspect the hardware: RAM, CPU, or motherboard problems, such as from \ndefects or overheating.\n\nI would suggest testing the machine outside of Java -- such as an \nintense memtest or CPU stress test -- to see if anything shows as flaky.\n\n- Gordon @ IA\n\n\n"}}